### PR TITLE
Add `SizeTrackingChannel`

### PR DESCRIPTION
**Pull request by chakaz**
This type of channel is very similar to `SimpleChannel`, but it also tracks the size of `T`s it owns.

### CODE REVIEW

**Review by romange — APPROVED** [review]
approved but how it helps you with measuring the memory usage?

**Comment by chakaz** [comment]
queue capacity * size of object is how much memory is used by the queue, am I missing something?

**Comment by chakaz** [comment]
I see what you may mean, I'll need to track each element's size as we save a string there as well :|

**Comment by romange** [comment]
If this object a string or any other object that uses heap then then the important part is missed. We do not care about 'sizeof string * capacity'

File: util/fibers/simple_channel.h [hunk 1]
```diff
@@ -126,6 +130,68 @@ template <typename T, typename Q> void SimpleChannel<T, Q>::StartClosing() {
   pop_ec_.notifyAll();
 }
 
+// SimpleQueue-like interface, but also keeps track over the size of Ts it owns.
+// It has a slightly less efficient TryPush() API as it forces construction of Ts even if they are
+// not pushed.
+// T must have a .size() method, which should return the heap-allocated size of T, excluding
+// anything included in sizeof(T). We could generalize this in the future.
+template <typename T, typename Queue = folly::ProducerConsumerQueue<T>> class SizeTrackingChannel {
```
Comment (romange) [inline util/fibers/simple_channel.h:138]: I prefer introducing this in the dragonfly codebase.

Comment (chakaz) [reply util/fibers/simple_channel.h]: Ok, will do.
Somewhat ironically, the only change I'll need to make is indeed the capacity() method then :)

**Pull request merged**